### DISCUSSION TITLE
Filter tabs before copyWindowInfo

### DISCUSSION THREAD

**Discussion by tamwile** [gdh1995/vimium-c#883] [Q&A] [answered]
Hi, when i use copyWindowInfo, it gives me all the tabs of the current window.
I would like to filter which tabs are passed to copyWindowInfo, either by domain name (choosing the same as the current tab), or by fuzzy finding (vomnibar maybe).

Is there a way to do this ?

**Comment by gdh1995** [comment, marked as answer]
Um like some other tab-related commands, `copyWindowInfo` support a parameter named `filter`, and:
* `filter` should be a string including a list of filter expressions joined with `+` or `&` character
* in your case, `map xxx copyWindowInfo filter="host"` should be enough
* there're so many expressions that I can't list them here, so if you're interested in it, please visit:

https://github.com/gdh1995/vimium-c/blob/a25ab8830f542b367d0be72578f59b641f4ac6cb/background/filter_tabs.ts#L180-L260

**Comment by tamwile** [comment]
How can i know which tab-related commands support a `filter` parameter ?

**Reply by gdh1995** [reply]
Um, no documents yet. You have to view the source code.
1. search for `filterTabsByCond_` in all `background/` files
2. find what commands the caller functions are by looking up the table of `availableCommands_` in `background/key_mappings.ts`.
